To: vim_dev@googlegroups.com
Subject: Patch 7.3.173
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.173
Problem:    After using setqflist() to make the quickfix list empty ":cwindow"
	    may open the window anyway.  Also after ":vimgrep".
Solution:   Correctly check whether the list is empty. (Ingo Karkat)
Files:	    src/quickfix.c


*** ../vim-7.3.172/src/quickfix.c	2011-04-28 17:27:05.000000000 +0200
--- src/quickfix.c	2011-05-05 16:55:47.000000000 +0200
***************
*** 1164,1170 ****
  
  	/* When no valid entries are present in the list, qf_ptr points to
  	 * the first item in the list */
! 	if (to_qfl->qf_nonevalid == TRUE)
  	    to_qfl->qf_ptr = to_qfl->qf_start;
      }
  
--- 1164,1170 ----
  
  	/* When no valid entries are present in the list, qf_ptr points to
  	 * the first item in the list */
! 	if (to_qfl->qf_nonevalid)
  	    to_qfl->qf_ptr = to_qfl->qf_start;
      }
  
***************
*** 2243,2248 ****
--- 2243,2249 ----
       * it if we have errors; otherwise, leave it closed.
       */
      if (qi->qf_lists[qi->qf_curlist].qf_nonevalid
+ 	    || qi->qf_lists[qi->qf_curlist].qf_count == 0
  	    || qi->qf_curlist >= qi->qf_listcount)
      {
  	if (win != NULL)
***************
*** 3711,3717 ****
      }
  
      if (qi->qf_lists[qi->qf_curlist].qf_index == 0)
! 	/* empty list or no valid entry */
  	qi->qf_lists[qi->qf_curlist].qf_nonevalid = TRUE;
      else
  	qi->qf_lists[qi->qf_curlist].qf_nonevalid = FALSE;
--- 3712,3718 ----
      }
  
      if (qi->qf_lists[qi->qf_curlist].qf_index == 0)
! 	/* no valid entry */
  	qi->qf_lists[qi->qf_curlist].qf_nonevalid = TRUE;
      else
  	qi->qf_lists[qi->qf_curlist].qf_nonevalid = FALSE;
*** ../vim-7.3.172/src/version.c	2011-05-05 16:41:19.000000000 +0200
--- src/version.c	2011-05-05 17:11:57.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     173,
  /**/

-- 
"I can't complain, but sometimes I still do."   (Joe Walsh)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
